mantidproject/mantid
# Slice Viewer ADS Observer

Introduces an ADS observer to the Slice Viewer so that when the workspace is modified outside the Viewer, it updates the plot with the new data.

**To test:**
1. Open a workspace in SliceViewer
2. Run an algorithm on the workspace, such as `scale`
3. Verify that the plot updates, along with the window title

Also verify conditions that should cause SliceViewer to close, either because you wouldn't be able to open a SliceViewer on that workspace in the first place, or because updating the view is too complex, such as: 
- Converting the workspace to a TableWorkspace
- Converting the workspace to any other type
- Converting a matrix workspace to a distribution
- Reducing the number of dimensions of a workspace to 1 or 0.

Fixes #29258 



---

#### Reviewer ####

Please comment on the following ([full description](http://developer.mantidproject.org/ReviewingAPullRequest.html)):

##### Code Review #####

- Is the code of an acceptable quality?
- Does the code conform to the [coding standards](http://developer.mantidproject.org/Standards/)?
- Are the unit tests small and test the class in isolation?
- If there is GUI work does it follow the [GUI standards](http://developer.mantidproject.org/Standards/GUIStandards.html)?
- If there are changes in the release notes then do they describe the changes appropriately?

##### Functional Tests #####

- Do changes function as described? Add comments below that describe the tests performed?
- Do the changes handle unexpected situations, e.g. bad input?
- Has the relevant (user and developer) documentation been added/updated?

Does everything look good? Mark the review as **Approve**. A member of `@mantidproject/gatekeepers` will take care of it.
